js/classes/entities/midground/chad.js

```diff
@@ -1,58 +1,373 @@
+        this.isOnGround = false;
+        /** Checks if CHAD can double jump. */
+        this.canDoubleJump = false;
+        /** Checks if CHAD has double jumped. */
+        this.hasDoubleJumped = false;
+        /** Checks if CHAD can dash. */
+        this.canDash = false;
+        /** Checks if CHAD has dashed. */
+        this.hasDashed = false;
+        /** Checks if CHAD is dashing. */
+        this.isDashing = false;
+        /** Gets the the y position of CHAD from the last time he was on the ground. */
+        this.prevYPosOnGround = 0;
+        /** The position of the Papa Chad (in the game world). */
         this.pos = pos;
-        /** An associative array of Chad's animations. Arranged [facing][action]. */
+        /** Gets the the x position of CHAD from the origin of where he started dashing. */
+        this.xDashAnchoredOrigin = 0;
+        /** An associative array of the animations for this Papa Chad. Arranged [facing][action]. */
         this.animations = [];
         this.loadAnimations();
-        /** What way is Chad looking? */
-        this.facing = "left"; // "left", "right"
-        /** What is Chad doing? */
+        /** What way is the Papa Chad looking? */
+        this.facing = "right";
+        /** What is the Papa Chad doing? */
         this.action = "idle";
         /** Used to check for collisions with other applicable entities. */
-        this.boundingBox = new BoundingBox();
+        this.boundingBox = new BoundingBox(this.pos, Chad.SCALED_SIZE);
         /** Used to check how to deal with collisions with other applicable entities. */
         this.lastBoundingBox = this.boundingBox;
+        /** The velocity at which Chad is moving. */
+        this.velocity = new Vector(0, 0);
+        /** Name of character. */
+        this.name = "dChad";
     };
 
-    /** The size, in pixels, of Chad's sprite ON THE SPRITESHEET. */
+    /** The size, in pixels of the sprite ON THE SPRITESHEET. */
     static get SIZE() {
+        return new Vector(29, 49);
+    }
 
-    };
-
-    /** How much bigger should Chad's sprite be drawn on the canvas than it is on the spritesheet? */
+    /** How much bigger should the sprite be drawn on the canvas than it is on the spritesheet? */
     static get SCALE() {
-
+        return 3;
     };
 
-    /** This will be Chad's size ON THE CANVAS. */
+    /** This will be the size of Papa Chad ON THE CANVAS. */
     static get SCALED_SIZE() {
         return Vector.multiply(Chad.SIZE, Chad.SCALE);
+    }
+
+    static get SPEED() {
+        return Chad.SCALE * 100;
     };
 
-    /** The filepath to Chad's spritesheet. */
+    /** The filepath to Papa Chad's spritesheet. */
     static get SPRITESHEET() {
-        return "./sprites/chad.js";
+        return "./sprites/parents.png";
     };
-    
-    /** Change what Chad is doing and where he is. */
+
+    /** The velocity of the first jump */
+    static get FIRST_JUMP_VELOCITY() {
```

Comment:
Semicolons in JS aren't SUPER important - in fact, they're often not necessary. But it is best practice i think to include the semicolons after each statement, as well as each class and function block.